### PR TITLE
Fix doctest + add quickcheck properties

### PR DESCRIPTION
**Pull request by timbuchwaldt**
This escalated quickly - I added 3 simple QuickCheck properties that at least discovered a small bug:
One could create a new node with an empty name, but never delete it from a HashRing again.

This also fixes a failing doctest that I had to get out of the way.

### CODE REVIEW

**Comment by bitwalker** [comment]
This is great! However we don't require node keys to be binary, so `byte_size` will fail for any keys which are non-binary. If we were going to require keys to be any type, it would be atom, since that's what node names are in the standard library anyway, but I would prefer not to go down that road. That said, I think what we can do is add a function head which handles the case where an empty binary is provided, and raise an argument error.

I love that you added QuickCheck tests! I've recently started doing that, and I'm a huge fan, so I appreciate you setting up the infrastructure :)

If you can make the adjustment I mentioned above, I'll get this merged ASAP!

**Comment by bitwalker** [comment]
Ping :)

**Comment by timbuchwaldt** [comment]
Hey,
sorry, I got caught up in vacation and stuff :D Will look into this now.

File: lib/ring.ex [hunk 1]
```diff
@@ -91,7 +91,10 @@ defmodule HashRing do
       "b"
   """
   @spec add_node(__MODULE__.t, term(), pos_integer) :: __MODULE__.t
-  def add_node(%__MODULE__{} = ring, node, weight \\ 128) when is_integer(weight) and weight > 0 do
+  def add_node(ring, node, weight \\ 128)
+  def add_node(_, node, _weight) when is_binary(node) and byte_size(node) == 0,
+    do: raise ArgumentError, message: "Node name needs to be longer than 0"
```
Comment (bitwalker) [inline lib/ring.ex]: Let's change this to `"Node keys cannot be empty strings"` - it's a little clearer I think. I can make this change myself too.

**Comment by timbuchwaldt** [comment]
There you go :)

**Comment by bitwalker** [comment]
Thank you sir!

**Pull request merged**